testsuite/tests/libratest/smoke_test.rs

```diff
@@ -695,3 +701,46 @@ fn test_full_node_basic_flow() {
         Decimal::from_str(&full_node_client_2.get_balance(&["b", "4"]).unwrap()).ok()
     );
 }
+
+#[test]
+fn test_e2e_reconfiguration() {
+    let (mut env, mut client_proxy_1) = setup_swarm_and_client_proxy(5, 1);
+    // the client connected to the removed validator
+    let mut client_proxy_0 = env.get_validator_ac_client(0);
+    client_proxy_1.create_next_account(false).unwrap();
+    client_proxy_0.set_accounts(client_proxy_1.copy_all_accounts());
+    client_proxy_1
+        .mint_coins(&["mintb", "0", "10"], true)
+        .unwrap();
+    assert_eq!(
+        Decimal::from_f64(10.0),
+        Decimal::from_str(&client_proxy_1.get_balance(&["b", "0"]).unwrap()).ok()
+    );
+    assert_eq!(
+        Decimal::from_f64(10.0),
+        Decimal::from_str(&client_proxy_0.get_balance(&["b", "0"]).unwrap()).ok()
+    );
+    let peer_id = env.get_validator(0).unwrap().validator_peer_id().unwrap();
+    client_proxy_1.remove_validator(peer_id, true).unwrap();
+    // mint another 10 coins after remove node 0
```

Comment:
Couple of dumb questions:

- Are we now injecting a `BlockMetadata` transaction at the beginning of every block? Without that, reconfiguration won't actually take effect, right?
- If yes to the question above, how do we know that the `mint` transaction below will be in a later block than the `remove_validator` transaction? If they're in the same block, the client connected to the removed validator will still see the update.

Perhaps we want to do something like asserting that `peer_id` is no longer a validator (based on a state read from `client_proxy1`) before encoding the mint transaction?

Reply:
This should not be a problem given we're doing blocking calls. 
We're not sending minting transaction until remove_validator is committed so the reconfiguration happens right after this remove_validator block, so the mint could only be included in next epoch.
But it's nice to have the functionality to assert membership.

Reply:
Got it, that makes sense. I forgot that blocking blocked until the transaction was included.

There is still the possibility that the transaction was included, but executing the transaction script failed (e.g., due to out-of-gas). But perhaps blocking can also protect us from this if it does something fancy like fail if the transaction is included with a status code other than `EXECUTED`?

Reply:
afaiu it only waits for the txn to be persisted not executed_state, but the assertion about amount would fail if the transaction failed.